src/test/scala/nl.knaw.dans.easy.deposit/docs/DDMSpec.scala

```diff
@@ -180,52 +183,116 @@ class DDMSpec extends TestSupportFixture with DdmBehavior {
+  }
+
+  "relation without qualifier" should "fail" in {
+    val url = "https://does.no.exist.dans.knaw.nl"
+    val relation = Relation(qualifier = None, Some(url), title = Some("blabla"))
+    DDM(new MinimalDatasetMetadata(relations = Some(Seq[RelationType](relation)))) should
+      beInvalidDoc(s"""invalid DatasetMetadata: no qualifier for Relation: {"url":"$url","title":"blabla"}""")
+  }
+
+  "spatial box without west" should "fail" in {
+    val box = """{"scheme":"x","north":"1","east":"2","south":"3"}"""
+    DDM(new MinimalDatasetMetadata(
+      spatialBoxes = parse(s"""{"spatialBoxes":[$box]}""").spatialBoxes
+    )) should beInvalidDoc(s"""invalid DatasetMetadata: not all of (north,east,south,west) provided by SpatialBox: $box""")
+  }
+
+  "spatial box without scheme" should "fail" in {
+    // note the differences in white spaces and quotes for the values
+    DDM(new MinimalDatasetMetadata(
+      spatialBoxes = parse("""{"spatialBoxes":[{"north": 1, "east": 2, "south": 3, "west": 4}]}""").spatialBoxes
+    )) should beInvalidDoc("""invalid DatasetMetadata: scheme is mandatory for SpatialBox: {"north":"1","east":"2","south":"3","west":"4"}""")
+  }
+
+  "spatial point without scheme" should "fail" in {
+    val point = """{"x":"1","y":"2"}"""
+    DDM(new MinimalDatasetMetadata(
+      spatialPoints = parse(s"""{"spatialPoints":[$point]}""").spatialPoints
+    )) should beInvalidDoc(s"""invalid DatasetMetadata: scheme is mandatory for SpatialPoint: $point""")
   }
 
-  "minimal with missing scheme for a SpatialPoint" should behave like validDatasetMetadata(
-    input = Try(new MinimalDatasetMetadata(spatialPoints = Some(Seq(
-      SpatialPoint(None, Some("1"), Some("2"))
-    ))))
+  "language key without scheme" should "fail" in {
+    DDM(new MinimalDatasetMetadata(
+      languagesOfFiles = parse("""{"languagesOfFiles":[ {"key":"NL","value":""}]}""").languagesOfFiles,
+    )) should beInvalidDoc(s"""invalid DatasetMetadata: SchemedKeyValue needs a scheme for a key {"key":"NL","value":""}""")
+  }
+
+  "language key with empty scheme" should "fail" in {
+    DDM(new MinimalDatasetMetadata(
+      languagesOfFiles = parse("""{"languagesOfFiles":[ {"scheme":" " "key":"NL","value":""}]}""").languagesOfFiles,
+    )) should beInvalidDoc(s"""invalid DatasetMetadata: SchemedKeyValue needs a scheme for a key {"scheme":" ","key":"NL","value":""}""")
+  }
+
+  "multiple datesCreated" should "fail" in {
+    the[IllegalArgumentException] thrownBy new DatasetMetadata(
+      dates = Some(mandatoryDates ++ mandatoryDates),
+    ) should have message
+      """requirement failed: At most one allowed; got [{"scheme":"dcterms:W3CDTF","value":"2018-05-29","qualifier":"dcterms:created"},{"scheme":"dcterms:W3CDTF","value":"2018-05-29","qualifier":"dcterms:created"}]"""
+  }
+
+  "metadata without datesCreated" should "fail" in {
+    DDM(new MinimalDatasetMetadata( // TODO DDM would not validate
```

Comment:
This test fails. As the UI is supposed to provided all mandatory values, this is not a direct production problem.